### DISCUSSION TITLE
Load Balancing: Query one upstream server at a time.. when does it try multiple?

### DISCUSSION THREAD

**Discussion by bcookatpcsd** [AdguardTeam/AdGuardHome#4023] [Q&A]
I've added NextDNS as recursive upstream.. and setup doh and dot urls.. 

In the Analytics tab of NextDNS.. AGH seems to *never* try the dot endpoint.. 

In the last 10 minutes since I've reset statistics.. 

![image](https://user-images.githubusercontent.com/55087301/147391308-e635c400-b9ce-49f2-a75e-680f2bf5f3d9.png)

I'm just not sure how it decides which is faster if the other is not used.. 

Is there anyway to do round robin?

**Comment by ZeroClover** [comment]
> Query one upstream server at a time. AdGuard Home uses its weighted random algorithm to pick the server so that the fastest server is used more often.

I think the DoH endpoint always return responses faster than DoT in the initial few queries, so AGH will be more aggressive in using DoH over DoT

**Comment by EugeneOne1** [comment]
@bcookatpcsd, hello.  Consider also checking the updated [wiki page](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file), especially the `all_servers` and `fastest_addr` settings description.
